coordinator/src/routes/health/get-status.ts

```diff
@@ -11,21 +12,45 @@ export interface StatusCheckResponse {
   links?: string
 }
 
-export async function serviceHealthCheck(url: string): Promise<StatusCheckResponse> {
-  return await axios
-    .get<string>(url, {timeout: 20000})
-    .then((response): StatusCheckResponse => ({
+export async function serviceHealthCheck(url: string, logger: pino.Logger): Promise<StatusCheckResponse> {
+  try {
+    const response = await axios.get<string>(url, {timeout: 20000})
+    return {
       status: response.status === 200 ? "pass" : "error",
       timeout: "false",
       responseCode: response.status,
       outcome: response.data,
       links: url
-    }))
-    .catch(error => ({
+    }
+  } catch (error) {
+    logger.error(error)
+    return {
       status: "error",
       timeout: error.code === "ECONNABORTED" ? "true" : "false",
       responseCode: 500
-    }))
+    }
+  }
+}
+
+function createStatusResponse(checks: { [p: string]: Array<StatusCheckResponse> }, h: Hapi.ResponseToolkit) {
+  let responseStatus = "pass"
+  let responseCode = 200
+  const warnFilter = Object.values(checks).flat().filter(response => response.status === "warn")
+  if (warnFilter.length > 0) {
```

Comment:
Rather than a filter followed by a length check, you could write this as:
```
const allChecks = Object.values(checks).flat()
if (allChecks.find(check => check.status === "warn")) {
...do warn stuff...
}
if (allChecks.find(check => check.status === "error")) {
...do warn stuff...
}
```

Reply:
done